packages/nextjs/src/server/utils.ts

```diff
@@ -234,3 +237,23 @@ export function assertTokenSignature(token: string, key: string, signature?: str
     throw new Error(authSignatureInvalid);
   }
 }
+
+/**
+ * Encrypt request data using signing key.
+ */
+export function encryptClerkRequestData(options: Partial<AuthenticateRequestOptions>): string {
+  return AES.encrypt(JSON.stringify(options), ENCRYPTION_KEY).toString();
```

Comment:
We should fallback to the secret key if no encryption key is provided. This makes it backwards compat and avoids the need for the most common use case to provide an `ENCRYPTION_KEY`.

```
if secretKey is provided as an option, then required ENCRYPTION_KEY
if no secretKey is provided as an option, ENCRYPTION_KEY is optional and falls back to SECRET_KEY
```